bridgenode/rev.go

```diff
@@ -62,19 +64,166 @@ func BlockAndRevReader(
+			curHeight++
+		}
+	}
+}
 
-		blockChan <- bnr
-		curHeight++
+// GetRawBlocksFromDisk retrives multiple consecutive blocks starting at height `startAt`.
+// `count` is a upper limit for the number of blocks read.
+// Only blocks that are contained in the same blk file are returned.
+func GetRawBlocksFromDisk(startAt int32, count int32, offsetFileName string,
+	blockDir string) (blocks []wire.MsgBlock, revs []RevBlock, err error) {
+	if startAt == 0 {
+		err = fmt.Errorf("Block 0 is not in blk files or utxo set")
+		return
+	}
+	startAt--
+
+	if count <= 0 {
+		return
+	}
+
+	offsetFile, err := os.Open(offsetFileName)
+	if err != nil {
+		return
+	}
+	defer offsetFile.Close() // file always closes
+
+	// offset file consists of 12 bytes per block
+	// tipnum * 12 gives us the correct position for that block
+	_, err = offsetFile.Seek(int64(12*startAt), 0)
+	if err != nil {
+		return
 	}
+	offsetReader := bufio.NewReaderSize(offsetFile, int(12*count))
+
+	offsets := make([]uint32, count)
+	revOffsets := make([]uint32, count)
+
+	var datFileNum uint32
+	// minOffset holds the smallest block offset that was seen
+	minOffset := uint32(math.MaxUint32)
+	// maxOffset holds the largest block offset that was seen
+	maxOffset := uint32(0)
+	// minRevOffset holds the smallest rev offset that was seen
+	minRevOffset := uint32(math.MaxUint32)
+	// maxRevOffset holds the largest rev offset that was seen
+	maxRevOffset := uint32(0)
+	// offsetsRead holds the number of blocks < count that are loacated in
+	// blk file with number `datFileNum`
+	offsetsRead := uint32(0)
+	for ; offsetsRead < uint32(count); offsetsRead++ {
+		var datFileNumTmp uint32
+		err = binary.Read(offsetReader, binary.BigEndian, &datFileNumTmp)
+		if err != nil {
+			break
+		}
+		if offsetsRead > 0 && datFileNumTmp != datFileNum {
+			// break if the block is located in a different blk file
+			break
+		}
+		datFileNum = datFileNumTmp
+		err = binary.Read(offsetReader, binary.BigEndian, &offsets[offsetsRead])
+		if err != nil {
+			return
+		}
+		err = binary.Read(offsetReader, binary.BigEndian, &revOffsets[offsetsRead])
+		if err != nil {
+			return
+		}
+
+		if offsets[offsetsRead] < minOffset {
+			minOffset = offsets[offsetsRead]
+		}
+		if offsets[offsetsRead] > maxOffset {
+			maxOffset = offsets[offsetsRead]
+		}
+
+		if revOffsets[offsetsRead] < minRevOffset {
+			minRevOffset = revOffsets[offsetsRead]
+		}
+		if revOffsets[offsetsRead] > maxRevOffset {
+			maxRevOffset = revOffsets[offsetsRead]
+		}
+	}
+
+	if offsetsRead == 0 {
+		return
+	}
+
+	blockFile, err := os.Open(filepath.Join(blockDir,
+		fmt.Sprintf("blk%05d.dat", datFileNum)))
+	if err != nil {
+		return
+	}
+	defer blockFile.Close()
+	blockFileInfo, _ := blockFile.Stat()
+
+	// Read all block data needed for the blocks into memory.
+	// The blocks that are needed are located between the minOffset and
+	// maxOffset+(the size of the block at maxOffset).
+	// 1<<23 = +8MB TODO: whats the max on disk size of a block?
```

Comment:
blk.dat files have a limit of 128MB. For us, we're really only gonna be dealing with 128MB blk files since we don't live sync.

I think it'd be better to have a constant of 128MB for the buffer rather than the current method in the code

Reply:
done. this is nice because now there is no need for the min/max offsets.